mdp/nodes/sfa_nodes.py

```diff
@@ -369,6 +376,170 @@ def get_eta_values(self, t=1):
+            There are several solving methods implemented:
+
+            reg  - works by regularization
+            pca  - works by PCA
+            svd  - works by SVD
+            ldl  - works by LDL decomposition (requires SciPy >= 1.0)
+
+            auto - (Will be: selects the best-benchmarked method of the above)
+                   Currently it simply selects pca.
+
+            Note: If you already received an exception
+            SymeigException ('Covariance matrices may be singular')
+            you can manually set the solving method for an existing node::
+
+               sfa.set_rank_deficit_method('pca')
+
+            That means,::
+
+               sfa = SFANode(rank_deficit='pca')
+
+            is equivalent to::
+
+               sfa = SFANode()
+               sfa.set_rank_deficit_method('pca')
+
+            After such an adjustment you can run ``stop_training()`` again,
+            which would save a potentially time-consuming rerun of all
+            ``train()`` calls.
+        :type rank_deficit_method: str
+        """
+        super(VartimeSFANode, self).__init__(input_dim,
+                                             output_dim, dtype, include_last_sample=True,
+                                             rank_deficit_method=rank_deficit_method)
+
+        self.tphase = 0
+
+    def _init_cov(self):
+        # init two covariance matrices
+        # one for the input data
+        self._cov_mtx = VartimeCovarianceMatrix(self.dtype)
+        # one for the derivatives
+        self._dcov_mtx = VartimeCovarianceMatrix(self.dtype)
+
+    def time_derivative(self, x, dt=None):
+        """
+        Compute the linear approximation of the time derivative
+
+        :param x: The time series data.
+        :type x: numpy.ndarray
+
+        :param dt: Sequence of time increments between vectors. Must be
+            of length *x.shape[0]-1* in case of a single training phase.
+            In case the training is done in multiple phases with intended
+            time dependence - meaning the derivative between chunks should
+            be considered, *dt* should be of length *x.shape[0]* starting
+            with the second call. If it continues to be of length
+            *x.shape[0]-1* time dependence is omitted. If *dt* is omitted
+            entirely there will be no time dependence between calls and
+            it will be considered to be one everywhere within a call.
+        :type dt: numpy.ndarray
+
+        :returns: Piecewise linear approximation of the time derivative.
+        :rtype: numpy.ndarray
+        """
+        outlen = x.shape[0] - \
+            1 if self.tphase == 0 or dt is None or dt.shape[0] != x.shape[0] else x.shape[0]
+        out = numx.empty([outlen, x.shape[1]])
+
+        if dt is not None:
+            # Improvements can be made, by interpolating polynomials
+            out[-x.shape[0]+1:, :] = (x[1:, :]-x[:-1, :]) / \
+                dt[-x.shape[0]+1:, None]
```

Comment:
In such a case I think it's best readable if you break the line after `=`. Also, please give the continuation line an additional indent to distinguish it from usual blocks.